src/player/PlayerComponent.cpp

```diff
@@ -1524,3 +1524,17 @@ QString PlayerComponent::videoInformation() const
   return infoStr;
 }
 
+/////////////////////////////////////////////////////////////////////////////////////////
+void PlayerComponent::setPlaybackSpeed(double speed)
+{
+  mpv::qt::set_property(m_mpv, "speed", speed);
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+double PlayerComponent::playbackSpeed()
+{
+  QVariant speed = mpv::qt::get_property(m_mpv, "speed");
+  if (speed.isValid())
+    return speed.toDouble();
+  return 0;
```

Comment:
C++ rookie question, the `speed` value set above is a `double`. But here the speed value returned in being handled as a `QVariant`. Is the treating `speed` as a `QVariant` just sugar to more easily ensure it's a non-null value?

What do you think of returning `1.0` when the speed is unknown? I'm concerned `0` is easy to conflate with a valid value. Additionally, if the UI has never called `setPlaybackSpeed` it seems reasonable that playback speed is a non-zero value (`1.0` being "normal").

Reply:
> C++ rookie question, the speed value set above is a double. But here the speed value returned in being handled as a QVariant. Is the treating speed as a QVariant just sugar to more easily ensure it's a non-null value?

That's right, `QVariant` is a type that holds another type, without enforcing a type at compile time and can be passed around without forcing a specific type.